sktime/forecasting/compose/_category_compositor.py

```diff
@@ -0,0 +1,252 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements a compositor to utilize forecasters based on ADI/CV categorization."""
+
+__author__ = ["shlok191"]
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.naive import NaiveForecaster
+from sktime.forecasting.sarimax import SARIMAX
+from sktime.forecasting.trend import PolynomialTrendForecaster
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.adi_cv import ADICVTransformer
+
+adi_cv_transformer = ADICVTransformer(features=["class"])
+
+
+class CategoryCompositor(BaseForecaster):
+    """Compositor that utilizes varying forecasters by time series data's nature.
+
+    Applies a series-to-primitives transformer on a given time series and utilizes
+    generated primitive value to apply the most appropriate forecaster to the
+    given series.
+
+    Parameters
+    ----------
+    forecasters : dict[sktime forecasters]
+        dict of forecasters with the key corresponding to categories generated
+        by the given transformer and the value corresponding to a sktime forecaster.
+
+    transformer : sktime transformer, default = ADICVTransformer()
+        A series-to-primitives sk-time transformer that generates a value
+        which can be used to quantify a choice of forecaster for the time series.
+
+    fallback_forecaster : sktime forecaster | None, Optional
+        A fallback forecaster that will be used if the category generated by
+        the transformer does not match any of the given forecasters.
+
+    Raises
+    ------
+    AssertionError: If a valid transformer (an instance of BaseTransformer)
+    is not passed or if valid forecasters (instances of BaseForecaster) is not given.
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.Series",
+        "scitype:y": "univariate",
+        "ignores-exogeneous-X": True,
```

Comment:
tags should be set differently - I would set scitype:y to both, ignores X to False; the inner mtypes to `pd.DataFrame` to handle multivariate data.

the capabilities, including "handles missing data" should be set so a capability is true if and only if all forecasters have the capability.

There is another thing we should do once this is sorted and tests pass, to handle global forecasters and hierarchical series - let´s make a note for it now, and I will explain it later once the simple case works.